lib/dialects/mysql/schema/columncompiler.js

```diff
@@ -4,6 +4,8 @@ const { inherits } = require('util');
 const ColumnCompiler = require('../../../schema/columncompiler');
 const { isObject } = require('../../../util/is');
 
+const commentEscapeRegex = /(?<!\\)'/g;
```

Comment:
Maybe this is normal for knex (e.g. to play it safe), but I wouldn't think it should concern itself with users pre-escaping their inputs to a particular dialect.  The interface, `comment(string)` is simple enough and should escape whatever is necessary to be escaped for the particular dialect.

Reply:
My quickfix for this bug was to escape the string myself. It's quite possible that other people have done the same and that would make their current code produce a different result. But I do agree with you, it shouldn't care if the quote is pre-escaped.

Reply:
@wickedest I'm concerned that otherwise this would have been a breaking change, since previously this functionality would only work if users escaped themselves, and that is no longer going to work.

Reply:
better to play it safe then

Reply:
hey this is broken for me.